sdk/src/androidTest/java/com/bugsnag/android/BreadcrumbsTest.java

```diff
@@ -172,4 +175,21 @@ public void testPayloadType() throws JSONException, IOException {
         assertEquals("left", node.getJSONObject("metaData").get("direction"));
         assertEquals(1, breadcrumbsJson.length());
     }
+
+    @Test
+    public void testClientMethods() {
+        Client client = generateClient();
+        client.leaveBreadcrumb("Hello World");
+        Queue<Breadcrumb> store = client.breadcrumbs.store;
+        int count = 0;
+
+        for (Breadcrumb breadcrumb : store) {
+            if (MANUAL == breadcrumb.getType() && "manual".equals(breadcrumb.getName())) {
+                count++;
+                assertEquals("Hello World", breadcrumb.getMetadata().get("message"));
+            }
+        }
+        assertEquals(1, count);
```

Comment:
Is the count necessary for this test? And if so, is the for loop the best way to test for it?  Would testing the length of the queue be more clear?

Reply:
The size and order of the breadcrumb queue cannot be guaranteed when calling `client#leaveBreadcrumb`, as automatic breadcrumbs may occur at any time (around 6-7 tend to be recorded in practice).